Library/Homebrew/extend/os/linux/dependency_collector.rb

```diff
@@ -0,0 +1,20 @@
+# typed: true
+# frozen_string_literal: true
+
+require "os/linux/glibc"
+
+class DependencyCollector
+  extend T::Sig
+
+  def gcc_dep_if_needed
+    return if DevelopmentTools.gcc_version("gcc") >= OS::LINUX_GCC_CI_VERSION
+
+    add(CompilerSelector.preferred_gcc, :build)
```

Comment:
Unsure when this should be build/not build.

Build handles the bottle case nicer but I feel pretty strongly we should just land this ASAP even if we end up with more GCC/glibc dependencies that necessary initially. People who care about this can clean it up.